allofplos/makedb.py

```diff
@@ -105,26 +110,33 @@ class PLOSArticle(BaseModel):
     word_count = IntegerField()
     JATS_type = ForeignKeyField(JATSType, related_name='jats')
 
+class SubjectsPLOSArticle(BaseModel):
+    subject = ForeignKeyField(Subjects)
+    article = ForeignKeyField(PLOSArticle)
+
 class CoAuthorPLOSArticle(BaseModel):
     corr_author = ForeignKeyField(CorrespondingAuthor)
     article = ForeignKeyField(PLOSArticle)
 
 db.connect()
-db.create_tables([Journal, PLOSArticle, ArticleType,
-                      CoAuthorPLOSArticle, CorrespondingAuthor,
-                      JATSType, Affiliations, Country])
+db.create_tables([Journal, PLOSArticle, ArticleType, CoAuthorPLOSArticle,
+                  CorrespondingAuthor, JATSType, Affiliations, Country,
+                  SubjectsPLOSArticle, Subjects])
+
 
 if args.starter:
-    allfiles = os.listdir('starter_corpus')
+    starter_dir = os.path.join(os.path.dirname(__file__), 'starter_corpus')
+    corpus = Corpus(starter_dir)
+    allfiles = corpus.iter_files
 else:
-    allfiles = os.listdir(corpusdir)
+    corpus = Corpus(corpusdir)
```

Comment:
defaults to `get_corpus_dir()` so you can just use `corpus = Corpus()`. only need to specify for `starterdir`

Reply:
ok, I called with get_corpus_dir because I read the init method of Corpus class and was None.
On a second inspection (after this comment), I realized that you are changing the corpusdir value when there is no value, so this is the same of declaring a default value, but with the problem that is not reflected in the method signature (you need the read the code to see it, when could be clear by just looking at the signature). So I am changing the init with this.
Looks at the changed init and tell me if I am missing something.